mobile/src/modules/transactions/screens/ViewQRCodeScreen.tsx

```diff
@@ -0,0 +1,249 @@
+import { convertNumberToNQTString } from '@burstjs/util';
+import React from 'react';
+import { Clipboard, Image, Share, StyleSheet, View } from 'react-native';
+import { NavigationInjectedProps, withNavigation } from 'react-navigation';
+import { connect } from 'react-redux';
+import { Button } from '../../../core/components/base/Button';
+import { Text, TextAlign } from '../../../core/components/base/Text';
+import { HeaderTitle } from '../../../core/components/header/HeaderTitle';
+import { i18n } from '../../../core/i18n';
+import { InjectedReduxProps } from '../../../core/interfaces';
+import { FullHeightView } from '../../../core/layout/FullHeightView';
+import { Screen } from '../../../core/layout/Screen';
+import { AppReduxState } from '../../../core/store/app/reducer';
+import { ApplicationState } from '../../../core/store/initialState';
+import { Colors } from '../../../core/theme/colors';
+import { defaultSideOffset, FontSizes, Sizes } from '../../../core/theme/sizes';
+import { EnterPasscodeModal } from '../../auth/components/passcode/EnterPasscodeModal';
+import { AuthReduxState } from '../../auth/store/reducer';
+import { ReceiveBurstPayload } from '../store/actions';
+import { TransactionsReduxState } from '../store/reducer';
+import { transactions } from '../translations';
+
+interface IProps extends InjectedReduxProps {
+  app: AppReduxState;
+  auth: AuthReduxState;
+  transactions: TransactionsReduxState;
+}
+
+type Props = ReceiveBurstPayload & IProps & NavigationInjectedProps;
+
+const styles = StyleSheet.create({
+  col: {
+    flex: 1
+  },
+  row: {
+    flex: 1,
+    display: 'flex',
+    flexDirection: 'row'
+  },
+  details: {
+    paddingHorizontal: defaultSideOffset,
+    paddingVertical: Sizes.MEDIUM,
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%'
+  },
+  imageWrapper: {
+    flex: 5
+  },
+  qrImage: {
+    width: 200,
+    height: 200,
+    alignSelf: 'center'
+  },
+  valueCol: {
+    flex: 3
+  },
+  buttonRow: {
+    flex: 2,
+    marginTop: 30
+  },
+  alignCenter: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    alignContent: 'center'
+  }
+});
+
+interface State {
+  isPINModalVisible: boolean;
```

Comment:
Don't think we need it everywhere. Probably should extract it to RootView someday later